rfd/0080-tls-ping.md

```diff
@@ -0,0 +1,415 @@
+it should add this prefix to the already existing MySQL protocol:
+`teleport-mysql-ping`.
+
+Ping protocols will take precedence over regular protocols. If the client asks
+for it and the proxy server supports, it will be used over the regular one.
+
+Clients and servers must rely on the `NegotiatedProtocol` connection state
+property to check if the protocol is supported on both sides and is going to be
+used for the connection.
+
+### Client/Server flow
+
+```mermaid
+sequenceDiagram
+    participant LP as Local Proxy (Client)
+    participant P as Proxy (Server)
+
+    LP->>P: Establish TLS Connection<br>(protocols = "teleport-sample-ping" and "teleport-sample")
+    
+    alt Ping protocol supported?
+        P->>LP: Connected<br>(negotiated protocol = "teleport-sample-ping")
+        P->>P: Starts using Ping connection
+        LP->>LP: Starts using Ping connection.
+        loop Every X time
+            P->>LP: Write Ping message
+        end
+    else
+        P->>LP: Connected<br>(negotiated protocol = "teleport-sample")
+    end
+
+    loop Continuosly
+        LP->P: Read/Write connection data
+    end
+```
+
+### UX
+
+#### Configuration
+
+Users will be able to configure the ping interval in their cluster
+configuration.
+
+```yaml
+auth_service:
+  # proxy_ping_interval defines in which interval the TLS routing ping message
+  # should be sent. This is applicable only when using ping-wrapped connections,
+  # regular TLS routing connections are not affected.
+  proxy_ping_interval: 30s
+```
+
+### Security
+
+#### Length overflow
+
+The connection will be responsible for casting the content length to the
+`length` type defined in the protocol packet. Depending on the size of the
+content, it can panic due to an overflow. If it happens on the client, the
+local proxy will be dropped, but in the Proxy server, it will cause the
+application to restart. To avoid this, the connection needs to limit the max
+`length` size.
+
+#### Broken connection
+
+There are some scenarios where the connection can be broken, meaning that `Read`
+calls might not return the expected content. All of those scenarios will only
+affect a single connection. The broken connection may or not affect the
+underlying protocols.
+
+##### Malformed message
+
+If the writer sends a message that doesn’t honor the packet or the data is not
+the same size defined by the `length` field, it will cause the `Read` calls to
+return wrong data, including the ping messages.
+
+##### Incomplete write
+
+Writing all data might require multiple write calls. If one fails, but the
+connection keeps going, it will cause bad reads since it expects a larger
+message than it was provided. In those scenarios, the best is to close the
+connection since there will be a high chance of it being broken.
+
+## Alternatives
+
+### HTTP2 Protocol
+
+Instead of introducing a custom protocol, we wrap the protocols on HTTP2 and
```

Comment:
Recently @r0mant found stack-overflow [issue](https://stackoverflow.com/questions/66818645/http2-ping-frames-over-aws-alb-grpc-keepalive-ping) where in case of ALB the http2 ping frame is handled by ALB so if we want to cover also ALB with h2 approach this seems to be a blocker for HTT2 protocol as a alternative.